src/network/networkd-route.c

```diff
@@ -2074,6 +2087,57 @@ int config_parse_route_type(
         return 0;
 }
 
+int config_parse_tcp_advmss(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(route_free_or_set_invalidp) Route *n = NULL;
+        Network *network = userdata;
+        uint64_t u;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = route_new_static(network, filename, section_line, &n);
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to allocate route, ignoring assignment: %m");
+                return 0;
+        }
+
+        r = parse_size(rvalue, 1024, &u);
```

Comment:
so, we should filter out "0" here, and refuse it.

Our usual way how to unset an option is by assigning the empty string to it. Hence, please:

1. refuse TCPAdvertisedMaximumSegmentSize=0 here
2. support TCPAdvertisedMaximumSegmentSize= and in that case set the field to 0

i.e. implicit behaviour of some magic undocumented special values is not OK. We should make this explicit, i.e. empty string should universally mean "unset" or "reset to defaults". And if you specify a numeric value, then the it should be taken literally, and if that makes no sense (like in this case afaics if zero) be refused.

Reply:
OK Updated. ptal. thanks .